src/DDTrace/Integrations/Curl/CurlIntegration.php

```diff
@@ -251,51 +291,61 @@ public function setup_curl_span($span) {
-        }
-
-        $span->resource = $normalizedPath;
-
-        /* Special case the Datadog Standard Attributes
-         * See https://docs.datadoghq.com/logs/processing/attributes_naming_convention/
-         */
-        if (!array_key_exists(Tag::HTTP_URL, $span->meta)) {
-            $span->meta[Tag::HTTP_URL] = $sanitizedUrl;
-        }
-
-        $span->peerServiceSources = HttpClientIntegrationHelper::PEER_SERVICE_SOURCES;
-
-        addSpanDataTagFromCurlInfo($span, $info, Tag::HTTP_STATUS_CODE, 'http_code');
-
-        addSpanDataTagFromCurlInfo($span, $info, 'network.client.ip', 'local_ip');
-        addSpanDataTagFromCurlInfo($span, $info, 'network.client.port', 'local_port');
-
-        addSpanDataTagFromCurlInfo($span, $info, 'network.destination.ip', 'primary_ip');
-        addSpanDataTagFromCurlInfo($span, $info, 'network.destination.port', 'primary_port');
-
-        addSpanDataTagFromCurlInfo($span, $info, 'network.bytes_read', 'size_download');
-        addSpanDataTagFromCurlInfo($span, $info, 'network.bytes_written', 'size_upload');
-
-        // Add the rest to a 'curl.' object
-        foreach ($info as $key => $val) {
-            // Datadog doesn't support arrays in tags
-            if (\is_scalar($val) && $val !== '') {
-                // Datadog sets durations in nanoseconds - convert from seconds
-                if (\substr_compare($key, '_time', -5) === 0) {
-                    $val *= 1000000000;
-                }
-                $span->meta["curl.{$key}"] = $val;
-            }
-        }
-
-        return $info;
-    }
+   public static function set_curl_attributes($span, $info) {
+       $sanitizedUrl = \DDTrace\Util\Normalizer::urlSanitize($info['url']);
+       $normalizedPath = \DDTrace\Util\Normalizer::uriNormalizeOutgoingPath($info['url']);
+       $host = Urls::hostname($sanitizedUrl);
+       $span->meta[Tag::NETWORK_DESTINATION_NAME] = $host;
+       unset($info['url']);
+
+       if (\dd_trace_env_config("DD_TRACE_HTTP_CLIENT_SPLIT_BY_DOMAIN")) {
+           $span->service = Urls::hostnameForTag($sanitizedUrl);
+       }
+
+       $span->resource = $normalizedPath;
+
+       /* Special case the Datadog Standard Attributes
+        * See https://docs.datadoghq.com/logs/processing/attributes_naming_convention/
+        */
+       if (!array_key_exists(Tag::HTTP_URL, $span->meta)) {
+           $span->meta[Tag::HTTP_URL] = $sanitizedUrl;
+       }
+
+       $span->peerServiceSources = HttpClientIntegrationHelper::PEER_SERVICE_SOURCES;
+
+       addSpanDataTagFromCurlInfo($span, $info, Tag::HTTP_STATUS_CODE, 'http_code');
+
+       // Check if the status code should be marked as an error based on configuration
+       if (isset($info['http_code']) && !empty($info['http_code']) && !isset($span->meta[Tag::ERROR])) {
+           $statusCode = (int)$info['http_code'];
+           if (self::isClientError($statusCode)) {
+               $span->meta[Tag::ERROR] = 1;
```

Comment:
```suggestion
       if (isset($info['http_code']) && !empty($info['http_code']) && !isset($span->meta[Tag::ERROR_MSG])) {
           $statusCode = (int)$info['http_code'];
           if (self::isClientError($statusCode)) {
```
Same for the other integrations - we're not supposed to set Tag::ERROR manually.